### PR TITLE
code cleanup: break up virt-handler's updateVMIStatus function

### PR DESCRIPTION
**Pull request by prnaraya**
**What this PR does / why we need it**: The updateVMIStatus has gotten quite long, so for maintainability and to reduce complexity, the function has been broken up. Since there were comments denoting the jobs of the various blocks of code, I moved these blocks into separate functions based on the way they were grouped up originally, as well as pulling out the code to handle sync errors and record an event when the VMI's phase changes into separate functions as well.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by vasiliy-ul — APPROVED** [review]
Looks good!

File: pkg/virt-handler/vm.go [hunk 1]
```diff
@@ -1263,6 +1263,36 @@ func (d *VirtualMachineController) updateSELinuxContext(vmi *v1.VirtualMachineIn
...
+	d.setMigrationProgressStatus(vmi, domain)
+	d.updateGuestInfoFromDomain(vmi, domain)
+	d.updateVolumeStatusesFromDomain(vmi, domain)
+	d.updateFSFreezeStatus(vmi, domain)
+	d.updateMachineType(vmi, domain)
+	err = d.netStat.UpdateStatus(vmi, domain)
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (fossedihelm) [inline pkg/virt-handler/vm.go:1281]: You can directly return `err`
```suggestion
        return err
```

**Review by fossedihelm — COMMENTED** [review]
Thanks! A small nit below

**Comment by fossedihelm** [comment]
/lgtm
Thanks

**Pull request merged**